### PR TITLE
Fix two potential memcpy of null pointers

### PR DESCRIPTION
**Pull request by pyhys**
Simplified logic by using multiple returns at error conditions.
Added error logging for invalid data pointers.

### CODE REVIEW

**Review by olbjo — COMMENTED** [review]
Added check is useful but one check should be enough. I might be missing something though.

File: src/device/pf_diag.c [hunk 1]
```diff
@@ -485,7 +493,10 @@ int pf_diag_add (
 
                p_item->usi = usi;
                p_item->fmt.usi.len = manuf_data_len;
-               memcpy (p_item->fmt.usi.manuf_data, p_manuf_data, manuf_data_len);
+               if (manuf_data_len > 0 && p_manuf_data != NULL)
```
[the remark this answers is not in the document]
Comment (pyhys) [reply src/device/pf_diag.c:496]: Changed to `if (manuf_data_len > 0)`
Done

File: src/device/pf_diag.c [hunk 2]
```diff
@@ -686,7 +705,10 @@ int pf_diag_update (
             {
                /* Manufacturer specific */
                p_item->fmt.usi.len = manuf_data_len;
-               memcpy (p_item->fmt.usi.manuf_data, p_manuf_data, manuf_data_len);
+               if (manuf_data_len > 0 && p_manuf_data != NULL)
```
[the remark this answers is not in the document]
Comment (pyhys) [reply src/device/pf_diag.c:708]: Changed to `if (manuf_data_len > 0)`
Done

**Review by olbjo — APPROVED** [review]
👍

**Comment by pyhys** [comment]
Made similar changes to `pf_alarm.c`

**Pull request merged**